### PR TITLE
Add Sign Up tests 8-13, 15-16

### PR DESCRIPTION
**Pull request by a-kotova**

### CODE REVIEW

File: cypress/fixtures/credentialsSignUp.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+  invalid: 'loripsum+2@@yahoo.com',
+};
+
+export const password = {
+  valid: 'qazwsx098',
+  short: 'q',
+};
+
+export const confirmPassword = {
+  equalToValidPassword: 'qazwsx098',
```
Comment (s34rching) [inline cypress/fixtures/credentialsSignUp.js]: you could just reuse `valid` <3

File: cypress/fixtures/credentialsSignUp.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+  equalToShortPassword: 'q',
+  unequalToPassword: 'qazwsx0988',
```
Comment (s34rching) [inline cypress/fixtures/credentialsSignUp.js]: no need, both of them are non equal to the initial, so they are equivalent

Comment (s34rching) [reply cypress/fixtures/credentialsSignUp.js]: anyway, if you are testing field validation of string length then it is ok

File: cypress/page-objects/signUpPage.js [hunk 3]
```diff
@@ -0,0 +1,73 @@
+import BasePage from './basePage';
```
Comment (s34rching) [inline cypress/page-objects/signUpPage.js]: I believe that the best solution was to combine `signIn` and `signUp` object methods and name the class and file something like `signingPage`, regarding the page is the same

File: package.json [hunk 4]
```diff
@@ -25,7 +25,7 @@
   },
   "devDependencies": {
     "concurrently": "^4.0.1",
-    "cypress": "6.2.0",
+    "cypress": "^7.7.0",
```
Comment (s34rching) [inline package.json]: you should never do this as a part of your ticket you are working on
trust me, bumping of packages version usually needs a lot of time and effort to be done
in our case: small app, small test suite - it is okay, but try to always create separate PR for this

File: cypress/integration/tests/signUp.js [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+    SignUpPage.signUp(name.valid, email.unregistered,
+      password.valid, confirmPassword.unequalToPassword);
+    cy.on('window:alert', (alertText) => {
+      expect(alertText).to.eq(errorMessages.passwordDoesntMatch);
+    });
+  });
+
+  it('TA-15: User unable to sign up with invalid Email', () => {
+    SignUpPage.signUp(name.valid, email.invalid,
+      password.valid, confirmPassword.equalToValidPassword);
```
Comment (s34rching) [inline cypress/integration/tests/signUp.js]: consider this syntax
```suggestion
    SignUpPage.signUp(
      name.valid,
      email.invalid,
      password.valid,
      confirmPassword.equalToValidPassword
    );
```
What you've done is really hard to catch

File: cypress/integration/tests/signUp.js [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+
+  it.skip('TA-17: User unable to sign up if enter whitespace in Name field', () => {
+    SignUpPage.signUp(' ', email.unregistered,
+      password.valid, confirmPassword.equalToValidPassword);
+    SignUpPage.signUpError.should('contain', errorMessages.nameConsistingOfWhitespaces);
+  });
+
+  it.skip('TA-18: User unable to sign up if enter whitespace in Password field', () => {
+    SignUpPage.signUp(name.valid, email.unregistered,
+      '       ', '       ');
```
Comment (s34rching) [inline cypress/integration/tests/signUp.js]: ?))) wtf?

**Review by s34rching — CHANGES_REQUESTED** [review]
My horse says "nay", so do I

File: package.json (hunk 4 above)
Comment (a-kotova) [reply package.json]: Discussed on call

File: cypress/fixtures/credentialsSignUp.js [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+};
+
+export const password = {
+  valid: 'qazwsx098',
+  short: 'q',
+  withWhitespace: ' qazwsx098',
+};
+
+export const confirmPassword = {
+  unequalToPassword: 'qazwsx0988',
```
Comment (s34rching) [inline cypress/fixtures/credentialsSignUp.js:18]: stick with `nonLikeSomething` or `notLikeSomething`

File: cypress/page-objects/signingPage.js [hunk 8]
```diff
@@ -0,0 +1,115 @@
...
+  }
+
+  signUp(name, email, password, confirmPassword) {
+    this.enterSignUpName(name);
+    this.enterSignUpEmail(email);
+    this.enterSignUpPassword(password);
+    this.enterSignUpConfirmPassword(confirmPassword);
+    this.submitSignUpForm();
+  }
+}
```
Comment (s34rching) [inline cypress/page-objects/signingPage.js]:
```suggestion
  signUp(name, email, password, confirmPassword) {
    if (name) {
      this.enterSignUpName(name);
    }
    if (email) {
      this.enterSignUpName(name);
    }
    if (password) {
      this.enterSignUpPassword(password);
    }
    if (confirmPassword) {
      this.enterSignUpConfirmPassword(confirmPassword);
    }
    this.submitSignUpForm();
  }
}
```

Take a look at this and think how you can reduce amount of lines for your tests while using conditionals and especially for tests with missing data

**Review by s34rching — CHANGES_REQUESTED** [review]

File: cypress/integration/tests/signUp.js [hunk 9]
```diff
@@ -0,0 +1,95 @@
...
+    SigningPage.signOutIcon.should('be.visible');
+  });
+
+  it('TA-8.1: User unable to sign up with already registered email', () => {
+    SigningPage.signUp(name.valid, email.registered, password.valid, password.valid);
+    SigningPage.signUpError.should('contain', errorMessages.alreadyRegisteredEmail);
+  });
+
+  it('TA-9: User unable to sign up with missing Name', () => {
+    SigningPage.signUp(undefined, email.registered, password.valid, password.valid);
```
Comment (s34rching) [inline cypress/integration/tests/signUp.js]: `null` is better, `null` means `no value`, `undefined` means `does not exists`

File: cypress/integration/tests/signUp.js [hunk 10]
```diff
@@ -0,0 +1,95 @@
...
+    SigningPage.emptySignUpNameField.should('have.length', 1);
+    SigningPage.emptySignUpNameField.then(($input) => {
+      expect($input[0].validationMessage).to.eq(errorMessages.emptyEmailOrPasswordOrNameField);
+    });
+  });
+
+  it('TA-10: User unable to sign up with missing Email', () => {
+    SigningPage.signUp(name.valid, undefined, password.valid, password.valid);
```
Comment (s34rching) [inline cypress/integration/tests/signUp.js]: `null` is better

File: cypress/integration/tests/signUp.js [hunk 11]
```diff
@@ -0,0 +1,95 @@
...
+    SigningPage.emptyOrInvalidSignUpEmailField.should('have.length', 1);
+    SigningPage.emptyOrInvalidSignUpEmailField.then(($input) => {
+      expect($input[0].validationMessage).to.eq(errorMessages.emptyEmailOrPasswordOrNameField);
+    });
+  });
+
+  it('TA-11: User unable to sigh up with missing Password', () => {
+    SigningPage.signUp(name.valid, email.registered, undefined, password.valid);
```
Comment (s34rching) [inline cypress/integration/tests/signUp.js]: `null` is better

File: cypress/integration/tests/signUp.js [hunk 12]
```diff
@@ -0,0 +1,95 @@
...
+    SigningPage.emptyPasswordField.should('have.length', 1);
+    SigningPage.emptyPasswordField.then(($input) => {
+      expect($input[0].validationMessage).to.eq(errorMessages.emptyEmailOrPasswordOrNameField);
+    });
+  });
+
+  it('TA-12: User unable to sign up with missing Confirm Password', () => {
+    SigningPage.signUp(name.valid, email.registered, password.valid, undefined);
```
Comment (s34rching) [inline cypress/integration/tests/signUp.js]: `null` is better

File: cypress/page-objects/signingPage.js [hunk 13]
```diff
@@ -0,0 +1,123 @@
...
+    return cy.get('input[data-test="sign-up-email"]:invalid');
+  }
+
+  get emptyPasswordField() {
+    return cy.get('input[data-test="sign-up-password"]:invalid');
+  }
+
+  get emptyConfirmPasswordField() {
+    return cy.get('input[data-test="sign-up-confirm-password"]:invalid');
+  }
```
Comment (s34rching) [inline cypress/page-objects/signingPage.js]: I'm not sure

File: cypress/page-objects/signingPage.js [hunk 14]
```diff
@@ -0,0 +1,123 @@
...
+    return cy.get('[data-test="signing-error"]');
+  }
+
+  get emptySignInEmailField() {
+    return cy.get('input[data-test="sign-in-email"]:invalid');
+  }
+
+  get emptySignInPasswordField() {
+    return cy.get('input[data-test="sign-in-password"]:invalid');
+  }
```
Comment (s34rching) [inline cypress/page-objects/signingPage.js]: I'm not sure

File: cypress/integration/tests/signUp.js [hunk 15]
```diff
@@ -0,0 +1,95 @@
...
+      expect(alertText).to.eq(errorMessages.passwordDoesntMatch);
+    });
+  });
+
+  it('TA-15: User unable to sign up with invalid Email', () => {
+    SigningPage.signUp(name.valid, email.invalid, password.valid, password.valid);
+    SigningPage.emptyOrInvalidSignUpEmailField.should('have.length', 1);
+    SigningPage.emptyOrInvalidSignUpEmailField.then(($input) => {
+      expect($input[0].validationMessage).to.eq(errorMessages.invalidEmailDoubledAtSign);
+    });
```
Comment (s34rching) [inline cypress/integration/tests/signUp.js]: I don't really the idea of adding `:invalid` to locators and creating `emptyOrInvalid...` objects.
What is on my mind: you don't need to exactly match the place of where this error happens but just the fact that the error message appears.
So please rework: assert that the field validation appeared in general on the form and particular error message is up.

**Review by s34rching — CHANGES_REQUESTED** [review]

**Review by s34rching — APPROVED** [review]

**Pull request merged**